### PR TITLE
Unicode Alphabet Support

### LINKED ISSUE #28: Option to allow unicode for AlphabeticValidator
### Description

The regex for the AlphabeticValidator does not allow unicode such as ÅÄÖ at the moment

### PR DESCRIPTION
**Pull request by ghost**
Resolves #28 `Option to allow unicode for AlphabeticValidator`.

### Summary

- Adds support for Unicode letters and numbers (as appropriate) to:
  - AlphabeticCondition
  - AlphanumericCondition
  - AlphabeticValidator
  - AlphanumericValidator
- Adds support for Whitespace to:
  - AlphanumericCondition
  - AlphanumericValidator
- Refactor validator tests to use new `AssertValidator` extension
- Turn off autocorrect in iOS Example

### Notes

1. This PR sets whether to allow Unicode letters and numbers or not to `false` by default (i.e. we don't allow them by default). This is to prevent a breaking API change. I feel that this should be reconsidered when we approach Version 2.0.
2. As email address validation is already complex, I have raised this as a separate issue (#33).
3. `NumericCondition` already support Unicode to some extent, as `NSRegularExpression` treats `\d` as any character in the Unicode General Category of Nd (see [Apple's documentation](https://developer.apple.com/reference/foundation/nsregularexpression#1965589)). I have raised a separate issue (#34) to make this match the others.

### CODE REVIEW

File: Tests/Unit Tests/Extensions/XCTestCase+Additions.swift [hunk 1]
```diff
@@ -25,4 +25,34 @@ extension XCTestCase {
...
+    ///
+    /// - Parameters:
+    ///   - validator: `Validator` to test.
+    ///   - testInput: Input to have `validator` check.
+    ///   - expectedResult: The expected result for running the validator's check.
+    ///   - file: File that the assertion is being run in. Defaults to `#file`.
+    ///   - line: Line that the assertion is being run from. Defaults to `#line`.
+    ///
+    /// - Note: Currently this checks that either the actual result and expected result are either both nil or both non-nil and have the same number of elements. As `Condition` currently does not conform to equatable, further checks are not done to ensure that the conditions in the array are the same.
+    func AssertValidator(_ validator: Validator, testInput: String?, expectedResult: [Condition]?, file: String = #file, line: UInt = #line) {
```
Comment (madhikarma) [inline Tests/Unit Tests/Extensions/XCTestCase+Additions.swift:38]: Just curious but we use this casing to indicate it's like a utility global function rather than an instance method?

Comment (madhikarma) [reply Tests/Unit Tests/Extensions/XCTestCase+Additions.swift:38]: I guess it we wanted it to be an action we would have called it `validate` or `verify` etc

File: Example/FormEntryView.swift [hunk 2]
```diff
@@ -45,6 +45,7 @@ final class FormEntryView<V: Validator>: UIView, ValidatorControlDelegate, UITex
         textLabel.textAlignment = .center
         stackView.addArrangedSubview(textLabel)
         
+        textField.autocorrectionType = .no
```
Comment (madhikarma) [inline Example/FormEntryView.swift:48]: Just curious but is this change related? My guess it to stop iOS from suggesting locale based suggestions for this internationalisation support or is there another reason?

File: Sources/Conditions/AlphabeticCondition.swift [hunk 3]
```diff
@@ -23,31 +23,34 @@ public struct AlphabeticCondition: Condition {
     
     public var shouldAllowViolation = true
     
+    /// Whether or not to allow Unicode letters. If `false` then only ASCII letters (A-Z, a-z) are allowed.
+    public let allowsUnicode: Bool
```
Comment (madhikarma) [inline Sources/Conditions/AlphabeticCondition.swift:27]: Bool is fine, I'm sure you've thought of this already that in future if the parameters increase we may require and configuration object or dictionary of parameters etc

I think it's fine for now and may never become a problem

**Review by madhikarma — APPROVED** [review]
Looks great @aamctustwo Just some questions / suggestions from me. Nice work

File: Example/FormEntryView.swift (hunk 2 above)
Comment (ghost) [reply Example/FormEntryView.swift:48]: I was have some issues running UI tests when it was enabled.

File: Tests/Unit Tests/Extensions/XCTestCase+Additions.swift (hunk 1 above)
Comment (ghost) [reply Tests/Unit Tests/Extensions/XCTestCase+Additions.swift:38]: Just legacy, to match the `AssertCondition` naming and make it internally consistent. I'm planning to do a full stylistic sweep before 2.0 to make sure everything is following the Swift 3 API design guidelines.

File: Sources/Conditions/AlphabeticCondition.swift (hunk 3 above)
Comment (ghost) [reply Sources/Conditions/AlphabeticCondition.swift:27]: Agreed.

Comment (ghost) [reply Sources/Conditions/AlphabeticCondition.swift:27]: Created a new issue for this: #36 `Update Conditions and Validators to Use Configs`.
